pkg/operator/ceph/cluster/nodedaemon/ceph-exporter.go

```diff
@@ -0,0 +1,241 @@
+		cephv1.GetCephExporterLabels(cephCluster.Spec.Labels).ApplyToObjectMeta(&deploy.ObjectMeta)
+		k8sutil.AddRookVersionLabelToDeployment(deploy)
+		if cephVersion != nil {
+			controller.AddCephVersionLabelToDeployment(*cephVersion, deploy)
+		}
+		deploy.Spec.Template = corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: deploymentLabels,
+			},
+			Spec: corev1.PodSpec{
+				NodeSelector: nodeSelector,
+				InitContainers: []corev1.Container{
+					getCephExporterChownInitContainer(cephCluster),
+				},
+				Containers: []corev1.Container{
+					getCephExporterDaemonContainer(cephCluster, *cephVersion),
+				},
+				Tolerations:   tolerations,
+				RestartPolicy: corev1.RestartPolicyAlways,
+				HostNetwork:   cephCluster.Spec.Network.IsHost(),
+				Volumes:       volumes,
+			},
+		}
+
+		return nil
+	}
+
+	return controllerutil.CreateOrUpdate(r.opManagerContext, r.client, deploy, mutateFunc)
+}
+
+func getCephExporterChownInitContainer(cephCluster cephv1.CephCluster) corev1.Container {
+	dataPathMap := config.NewDatalessDaemonDataPathMap(cephCluster.GetNamespace(), cephCluster.Spec.DataDirHostPath)
+	configDir := path.Join(cephCluster.Spec.DataDirHostPath, cephCluster.Namespace)
+	mounts := append(controller.DaemonVolumeMounts(dataPathMap, "", cephCluster.Spec.DataDirHostPath), v1.VolumeMount{Name: "ceph-conf-dir", MountPath: configDir})
+
+	return controller.ChownCephDataDirsInitContainer(
+		*dataPathMap,
+		cephCluster.Spec.CephVersion.Image,
+		controller.GetContainerImagePullPolicy(cephCluster.Spec.CephVersion.ImagePullPolicy),
+		mounts,
+		cephv1.GetCephExporterResources(cephCluster.Spec.Resources),
+		controller.PodSecurityContext(),
+	)
+}
+
+func getCephExporterDaemonContainer(cephCluster cephv1.CephCluster, cephVersion cephver.CephVersion) corev1.Container {
+	cephImage := cephCluster.Spec.CephVersion.Image
+	dataPathMap := config.NewDatalessDaemonDataPathMap(cephCluster.GetNamespace(), cephCluster.Spec.DataDirHostPath)
+	configDir := path.Join(cephCluster.Spec.DataDirHostPath, cephCluster.Namespace)
+	cephConfFile := path.Join(configDir, fmt.Sprintf("%s.config", cephCluster.Namespace))
+	volumeMounts := append(controller.DaemonVolumeMounts(dataPathMap, "", cephCluster.Spec.DataDirHostPath), v1.VolumeMount{Name: "ceph-conf-dir", MountPath: configDir})
+
+	container := corev1.Container{
+		Name:            "ceph-exporter",
+		Command:         []string{"/usr/bin/ceph-exporter"},
+		Args:            []string{"--conf", cephConfFile, "--sock-dir", sockDir, "--port", strconv.Itoa(int(DefaultMetricsPort)), "--prio-limit", perfCountersPrioLimit, "--stats-period", statsPeriod},
+		Image:           cephImage,
+		ImagePullPolicy: controller.GetContainerImagePullPolicy(cephCluster.Spec.CephVersion.ImagePullPolicy),
+		VolumeMounts:    volumeMounts,
+		Resources:       cephv1.GetCephExporterResources(cephCluster.Spec.Resources),
+		SecurityContext: controller.PodSecurityContext(),
```

Comment:
Let's add the priority class similar to the crash collector, and you'll need to define this method and also add an example of this priority class in the cluster.yaml example [here](https://github.com/rook/rook/blob/master/deploy/examples/cluster.yaml#L219-L224):
```go
PriorityClassName: cephv1.GetExporterPriorityClassName(cephCluster.Spec.PriorityClassNames),
```
This should fix the unit test issue in this package